kafka_topic_enforcer/src/main/java/com/tesla/data/topic/enforcer/EnforceCommand.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019. Tesla Motors, Inc. All rights reserved.
+ */
+
+package com.tesla.data.topic.enforcer;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.exporter.HTTPServer;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.KafkaAdminClient;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Supplier;
+
+@Parameters(commandDescription = "Enforce given configuration")
+public class EnforceCommand extends BaseCommand {
+
+  private static final Gauge lastSuccess = Gauge.build()
+      .name("kafka_topic_enforcer_run_last_success_timestamp")
+      .help("Timestamp of last successful run as epoch seconds.")
+      .register();
+  private static final Counter failures = Counter.build()
+      .name("kafka_topic_enforcer_run_failures_total")
+      .help("Total failures so far.")
+      .register();
+  private static final Counter runs = Counter.build()
+      .name("kafka_topic_enforcer_run_total")
+      .help("Total runs so far.")
+      .register();
+
+  @Parameter(
+      names = {"--dryrun", "-d"},
+      description = "do a dry run")
+  private boolean dryrun = false;
+  @Parameter(
+      names = {"--unsafemode"},
+      description = "run in unsafe mode, topic deletion is _only_ allowed in this mode")
+  private boolean unsafemode = false;
+  @Parameter(
+      names = {"--continuous", "-c"},
+      description = "run enforcement continuously")
+  private boolean continuous = false;
+  @Parameter(
+      names = {"--interval", "-i"},
+      description = "run interval for continuous mode in seconds",
+      arity = 1)
+  private int interval = 600;
+
+  EnforceCommand() {
+    // DO NOT REMOVE, this is needed by jcommander
+  }
+
+  // for testing
+  EnforceCommand(Map<String, Object> cmdConfig, boolean continuous, int interval) {
+    super(cmdConfig);
+    this.continuous = continuous;
+    this.interval = interval;
+  }
+
+  @Override
+  public int run() {
+    try (AdminClient kafka = KafkaAdminClient.create(kafkaConfig())) {
+      TopicService topicService = new TopicServiceImpl(kafka, dryrun);
+      Enforcer enforcer = new Enforcer(topicService, configuredTopics(), !unsafemode);
+      return doRun(enforcer, () -> false);
```

Comment:
maybe just make this `() -> !continuous` and make the while a do-while, and then you just have a single code path. Spins up the prom server extra, but is less complexity.

Worth thinking about at least, not blocking comment.

Reply:
Good point. There is a sleep as well, I cleaned up the method a bit to improve readability.